crates/typst-pdf/src/page.rs

```diff
@@ -263,6 +278,56 @@ pub struct Page {
     pub links: Vec<(Destination, Rect)>,
     /// The page's PDF label.
     pub label: Option<PdfPageLabel>,
+    /// The page's used resources
+    pub resources: HashMap<PageResource, usize>,
+}
+
+/// Represents a resource being used in a PDF page by its name.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum PageResource {
+    XObject(EcoString),
```

Comment:
should this enum really have the names or should it be a plain enum and the name is stored separately? also, do the names already contain the letter prefixes? if so, it feels like the information is somewhat redundant.

Reply:
The information is important to be able to write the resource dictionaries (which are per-resource). But you're right, I'll change it to have the name stored alongside it